### PR TITLE
[zziplib] Uses upstream CMakeLists.txt

### LINKED ISSUE #15203: [zziplib] x64-mingw-dynamic build failure
**Host Environment**
- OS: Windows
- Compiler: Mingw-w64 8.1.0 as provided by [Github Actions](https://github.com/actions/virtual-environments/blob/main/images/win/Windows2019-Readme.md).

**To Reproduce**
`./vcpkg install zziplib`

**Failure logs** 
```
  Starting package 2/2: zziplib:x64-mingw-dynamic
  Building package zziplib[core]:x64-mingw-dynamic...
  -- Using community triplet x64-mingw-dynamic. This triplet configuration is not guaranteed to succeed.
  -- [COMMUNITY] Loading triplet configuration from: D:\a\Mudlet\Mudlet\3rdparty\vcpkg\triplets\community\x64-mingw-dynamic.cmake
  -- Downloading https://github.com/gdraheim/zziplib/archive/24a6c6de1956189bffcd8dffd2ef3197c6f3df29.tar.gz...
  -- Extracting source D:/a/Mudlet/Mudlet/3rdparty/vcpkg/downloads/gdraheim-zziplib-24a6c6de1956189bffcd8dffd2ef3197c6f3df29.tar.gz
  -- Using source at D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean
  -- Configuring x64-mingw-dynamic-dbg
  -- Configuring x64-mingw-dynamic-rel
  -- Building x64-mingw-dynamic-dbg
  CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:145 (message):
      Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v -j3
      Working Directory: D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/x64-mingw-dynamic-dbg
      See logs for more information:
        D:\a\Mudlet\Mudlet\3rdparty\vcpkg\buildtrees\zziplib\install-x64-mingw-dynamic-dbg-out.log
  
  Call Stack (most recent call first):
    scripts/cmake/vcpkg_build_cmake.cmake:96 (vcpkg_execute_build_process)
    scripts/cmake/vcpkg_install_cmake.cmake:27 (vcpkg_build_cmake)
    ports/zziplib/portfile.cmake:27 (vcpkg_install_cmake)
    scripts/ports.cmake:136 (include)
  
  
  Error: Building package zziplib:x64-mingw-dynamic failed with: BUILD_FAILED
  Please ensure you're using the latest portfiles with `.\vcpkg update`, then
  submit an issue at https://github.com/Microsoft/vcpkg/issues including:
    Package: zziplib:x64-mingw-dynamic
    Vcpkg version: 2020.11.12-nohash
  
  Additionally, attach any relevant sections from the log files above.
  Error: Last command execution failed with error code '1'.
```

```
install-x64-mingw-dynamic-dbg-out.log: 
[1/10] C:\ProgramData\chocolatey\bin\x86_64-w64-mingw32-gcc.exe -DZZIPLIB_EXPORTS -Dzziplib_EXPORTS -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-mingw-dynamic/include -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean -g -MD -MT CMakeFiles/zziplib.dir/zzip/file.c.obj -MF CMakeFiles\zziplib.dir\zzip\file.c.obj.d -o CMakeFiles/zziplib.dir/zzip/file.c.obj -c D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/file.c
FAILED: CMakeFiles/zziplib.dir/zzip/file.c.obj 
C:\ProgramData\chocolatey\bin\x86_64-w64-mingw32-gcc.exe -DZZIPLIB_EXPORTS -Dzziplib_EXPORTS -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-mingw-dynamic/include -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean -g -MD -MT CMakeFiles/zziplib.dir/zzip/file.c.obj -MF CMakeFiles\zziplib.dir\zzip\file.c.obj.d -o CMakeFiles/zziplib.dir/zzip/file.c.obj -c D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/file.c
In file included from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/types.h:25,
                 from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/zzip.h:17,
                 from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/lib.h:16,
                 from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/file.c:10:
D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/conf.h:33:11: fatal error: zzip/_config.h: No such file or directory
 # include <zzip/_config.h>
           ^~~~~~~~~~~~~~~~
compilation terminated.
[2/10] C:\ProgramData\chocolatey\bin\x86_64-w64-mingw32-gcc.exe -DZZIPLIB_EXPORTS -Dzziplib_EXPORTS -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-mingw-dynamic/include -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean -g -MD -MT CMakeFiles/zziplib.dir/zzip/dir.c.obj -MF CMakeFiles\zziplib.dir\zzip\dir.c.obj.d -o CMakeFiles/zziplib.dir/zzip/dir.c.obj -c D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/dir.c
FAILED: CMakeFiles/zziplib.dir/zzip/dir.c.obj 
C:\ProgramData\chocolatey\bin\x86_64-w64-mingw32-gcc.exe -DZZIPLIB_EXPORTS -Dzziplib_EXPORTS -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-mingw-dynamic/include -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean -g -MD -MT CMakeFiles/zziplib.dir/zzip/dir.c.obj -MF CMakeFiles\zziplib.dir\zzip\dir.c.obj.d -o CMakeFiles/zziplib.dir/zzip/dir.c.obj -c D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/dir.c
In file included from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/types.h:25,
                 from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/zzip.h:17,
                 from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/lib.h:16,
                 from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/dir.c:9:
D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/conf.h:33:11: fatal error: zzip/_config.h: No such file or directory
 # include <zzip/_config.h>
           ^~~~~~~~~~~~~~~~
compilation terminated.
[3/10] C:\ProgramData\chocolatey\bin\x86_64-w64-mingw32-gcc.exe -DZZIPLIB_EXPORTS -Dzziplib_EXPORTS -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-mingw-dynamic/include -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean -g -MD -MT CMakeFiles/zziplib.dir/zzip/fetch.c.obj -MF CMakeFiles\zziplib.dir\zzip\fetch.c.obj.d -o CMakeFiles/zziplib.dir/zzip/fetch.c.obj -c D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/fetch.c
FAILED: CMakeFiles/zziplib.dir/zzip/fetch.c.obj 
C:\ProgramData\chocolatey\bin\x86_64-w64-mingw32-gcc.exe -DZZIPLIB_EXPORTS -Dzziplib_EXPORTS -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/installed/x64-mingw-dynamic/include -ID:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean -g -MD -MT CMakeFiles/zziplib.dir/zzip/fetch.c.obj -MF CMakeFiles\zziplib.dir\zzip\fetch.c.obj.d -o CMakeFiles/zziplib.dir/zzip/fetch.c.obj -c D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/fetch.c
In file included from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/types.h:25,
                 from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/fetch.h:4,
                 from D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/fetch.c:13:
D:/a/Mudlet/Mudlet/3rdparty/vcpkg/buildtrees/zziplib/src/97c6f3df29-5eedf12ac5.clean/zzip/conf.h:33:11: fatal error: zzip/_config.h: No such file or directory
 # include <zzip/_config.h>
           ^~~~~~~~~~~~~~~~
compilation terminated.
ninja: build stopped: subcommand failed.
```

### PR DESCRIPTION
**Pull request by longnguyen2004**
**Describe the pull request**

- What does your PR fix? Fixes #15203
This PR removes the current CMakeLists.txt included with the zziplib port, and uses the upstream one.
- Which triplets are supported/not supported? Have you updated the CI baseline?
All triplets
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

### CODE REVIEW

**Comment by longnguyen2004** [comment]
Can someone get osx build logs for me?

**Comment by NancyLi1013** [comment]
Please see the details here
https://dev.azure.com/vcpkg/public/_build/results?buildId=46773&view=artifacts&pathAsName=false&type=publishedArtifacts

Since there is something wrong on osx, we cannot get logs on osx now.

The failures on Linux:

```
CMake Error at zzip/CMakeLists.txt:315 (get_target_property):
  get_target_property() called with non-existent target "libzzipfseeko".


CMake Error at zzip/CMakeLists.txt:316 (get_target_property):
  get_target_property() called with non-existent target "libzzipfseeko".


CMake Error at zzip/CMakeLists.txt:317 (get_target_property):
  get_target_property() called with non-existent target "libzzipfseeko".


CMake Error at zzip/CMakeLists.txt:330 (get_target_property):
  get_target_property() called with non-existent target "libzzipmmapped".


CMake Error at zzip/CMakeLists.txt:331 (get_target_property):
  get_target_property() called with non-existent target "libzzipmmapped".


CMake Error at zzip/CMakeLists.txt:332 (get_target_property):
  get_target_property() called with non-existent target "libzzipmmapped".

```

The failures on uwp:

```
4>D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip\err.c(70,20): error C4996: 'strerror': This function or variable may be unsafe. Consider using strerror_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [D:\buildtrees\zziplib\x64-uwp-dbg\zzip\libzzip.vcxproj]
     4>D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip/__dirent.h(174,12): error C4996: 'strcpy': This function or variable may be unsafe. Consider using strcpy_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. (compiling source file D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip\dir.c) [D:\buildtrees\zziplib\x64-uwp-dbg\zzip\libzzip.vcxproj]
     4>D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip\err.c(75,12): error C4996: 'strerror': This function or variable may be unsafe. Consider using strerror_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [D:\buildtrees\zziplib\x64-uwp-dbg\zzip\libzzip.vcxproj]
     4>D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip/__dirent.h(180,9): error C4996: 'strcat': This function or variable may be unsafe. Consider using strcat_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. (compiling source file D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip\dir.c) [D:\buildtrees\zziplib\x64-uwp-dbg\zzip\libzzip.vcxproj]
     4>D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip\err.c(87,16): error C4996: 'strerror': This function or variable may be unsafe. Consider using strerror_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [D:\buildtrees\zziplib\x64-uwp-dbg\zzip\libzzip.vcxproj]
     4>D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip/__dirent.h(184,12): error C4996: 'strcat': This function or variable may be unsafe. Consider using strcat_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. (compiling source file D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip\dir.c) [D:\buildtrees\zziplib\x64-uwp-dbg\zzip\libzzip.vcxproj]
     4>D:\buildtrees\zziplib\src\97c6f3df29-1240d2902f.clean\zzip\plugin.c(45,6): error C4996: 'open': The POSIX name for this item is deprecated. Instead, use the ISO C and C++ conformant name: _open. See online help for details. [D:\buildtrees\zziplib\x64-uwp-dbg\zzip\libzzip.vcxproj]
```

File: ports/zziplib/portfile.cmake [hunk 1]
```diff
@@ -3,25 +3,45 @@ vcpkg_from_github(
     REPO gdraheim/zziplib
     REF 24a6c6de1956189bffcd8dffd2ef3197c6f3df29 # v0.13.71
     SHA512 246ee1d93f3f8a6889e9ab362e04e6814813844f2cdea0a782910bf07ca55ecd6d8b1c456b4180935464cebf291e7849af27ac0ed5cc080de5fb158f9f3aeffb
+    PATCHES
+        install-dll-to-proper-folder.patch
+        no-release-postfix.patch
+        fix-export-define.patch
+        suppress-c4996-msvc.patch
+        always-find-unixcommands-on-unix.patch
```
Comment (NancyLi1013) [inline ports/zziplib/portfile.cmake:11]: Is it possible to submit these patches to upstream?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/zziplib/portfile.cmake (hunk 1 above)
Comment (longnguyen2004) [reply ports/zziplib/portfile.cmake:11]: I'm currently submitting it upstream. `fix-export-define.patch` is already submitted.

**Comment by vadi2** [comment]
Can you merge latest `development`? It fails on building the dependency `zlib` before it can get to `zziplib`.

**Comment by longnguyen2004** [comment]
@vadi2 Can you give me the build log?

**Comment by vadi2** [comment]
Yeah.

**Review by vadi2 — APPROVED** [review]
Builds OK

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for getting us to use upstream things! :D

**Pull request merged**